modules/sql-engine/src/main/codegen/config.fmpp

```diff
@@ -57,85 +57,47 @@ data: {
     # not a reserved keyword, add it to the 'nonReservedKeywords' section.
     keywords: [
       "COLOCATE"
-      "SEMI"
       "IF"
       "INDEX"
       "ENGINE"
-#     "KEY_TYPE" // already presented in Calcite
       "TREE"
       "HASH"
-      "RENAME",
-      "UUID",
+      "RENAME"
+      "UUID"
     ]
 
     # List of non-reserved keywords to add;
-    # items in this list become non-reserved
-    nonReservedKeywords: [
-      "SEMI"
+    # items in this list become non-reserved.
+    nonReservedKeywordsToAdd: [
```

Comment:
This patch changes the configuration of FMPP task in order to take an advantage of a default configuration introduced in [CALCITE-4238](https://issues.apache.org/jira/browse/CALCITE-4238). Having default configuration, any sub parser (as Ignite parser, for instance) may omit redundant configuration, and concentrate only on extensions to the grammar. One of such extensions is a list of keywords that should not be reserved. That is, keywords that can be freely used as identifiers of a table/column/index/whatever without need to be quoted.

A "nonReservedKeywordsToAdd" is an extension point to convert any keyword to non-reserved. Any keyword from this list, as well as any keyword from `default_config.fmpp#nonReservedKeywords`, can be used as an identifier without quoting.

Reply:
thanks, got it ! But i have strange feeling that we can miss here something, look for example :
CASCADE is reserved by standard but ignite allow to create a table with such a name, can we obtain a troubles in future if we want to process this keyword ? 
also i find that errors with existing reserved keywords are non informative :
_CREATE TABLE INDEX_ brings :
```
Caused by: org.apache.ignite.internal.generated.query.calcite.sql.ParseException: Incorrect syntax near the keyword 'TABLE' at line 1, column 8.
Was expecting one of:
    "OR" ...
    "TABLE" ...
```
i suppose we need additional issue here, wdyt?

Reply:
> CASCADE is reserved by standard 

I haven't found segregation on reserved vs non-reserved keywords in SQL standard (to be honest, I've spent not much time on this). If you know particular section, I appreciate if you point it out to me

> i suppose we need additional issue here, wdyt?

IN fact, we already have such issue. I'm going to address error message in [IGNITE-20104](https://issues.apache.org/jira/browse/IGNITE-20104)

Reply:
5.2 <token> and <separator>
section

Reply:
wow, that's cool, thanks!

I tried to search for `reserved` word, but search results were pointing out on a random place at document. Turns out search just got broken because of there were too much options since every page contains `All rights reserved` at the bottom...

regarding to `CASCADE`: according to SQL standard 2016, CASCADE is non reserved.

regarding to `can we obtain a troubles in future if we want to process this keyword ?`: technically, yes. But the list of non reserved keywords is aligned with Postgres and H2, thus I believe there won't be any problem

Reply:
I check 2 random words form 2016 standard and it`s passed : SYSTEM_TIME, TRUNCATE probably we need additional issue ? btw they are passed on main too. Plz fix conflicts and move forward.

Reply:
> I check 2 random reserved words form 2016 standard and it`s passed

I'm going to revise list of (non-)reserved keywords in IGNITE-20104, thus I believe it's better to continue this discussion in the next patch :)